src/lib/y2partitioner/widgets/encrypt_password.rb

```diff
@@ -93,6 +73,94 @@ def pw1
       def pw2
         Yast::UI.QueryWidget(Id(:pw2), :Value)
       end
+
+      # @return [Boolean]
+      def random_password_allowed?
+        @random_password_allowed ||= @controller.filesystem_type.to_sym == :swap
```

Comment:
please add explanation why random password is allowed only for swap. Ideally link to some documentation or short explanation with googable content.

Reply:
OK, the presence of this method answers my earlier question :)
@jreidinger : I don't think we need to link anything, the reason is simple: what happens when you let YaST use a random password for /home? (or any other non-swap FS). But yes, a short comment is needed.

Reply:
@dgdavid The `||=` method of memoization does not work well when the value is a Boolean. I think a `.nil?` is needed.

Reply:
@mvidner well, still I would like to have written down this reason, as not everyone knows about this and there are other devices where it can make sense like e.g. bcache

Reply:
@mvidner,

Oops! Yes, you are right. _It's not good to be in a hurry_ :innocent: 

This is, probably, the first method that I wrote when started to play with the bug a few days ago. I think that since filesystem does not will change and right now we are considering the swap as the only filesystem that allows this kind of encryption method, it could be declared in the `#initialize` method. Isn't it?

My doubt is about the predicate method. Should I to add a method, use an alias_method or simply forget them and use directly the variable itself?

I mean

```ruby
...

def random_password_allowed?
  @random_password_allowed
end
```

or

```ruby

...

private

...
attr_reader :random_password_allowed
alias_method :random_password_allowed?, :random_password_allowed
...
```

or simply use the variable

```ruby
...

do_something if random_password_allowed
...
private

attr_reader :random_password_allowed
...

```